client/extensions/woocommerce/app/settings/payments/payment-method-item.js

```diff
@@ -1,76 +1,163 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import Button from 'components/button';
+import {
+	changePaymentMethodField,
+	closeEditingPaymentMethod,
+	openPaymentMethodForEdit,
+} from 'woocommerce/state/ui/payments/methods/actions';
+import { getCurrentlyEditingPaymentMethod } from 'woocommerce/state/ui/payments/methods/selectors';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { errorNotice, successNotice } from 'state/notices/actions';
 import ListItem from 'woocommerce/components/list/list-item';
 import ListItemField from 'woocommerce/components/list/list-item-field';
+import PaymentMethodEdit from './payment-method-edit';
+import { savePaymentMethod } from 'woocommerce/state/sites/payment-methods/actions';
 
-const PaymentMethodItem = ( { method, translate, onCancel, onEdit, currentlyEditingId } ) => {
-	let editButtonText = translate( 'Set up' );
-	if ( currentlyEditingId === method.id ) {
-		editButtonText = translate( 'Cancel' );
-	}
+class PaymentMethodItem extends Component {
+	static propTypes = {
+		closeEditingPaymentMethod: PropTypes.func.isRequired,
+		currentlyEditingId: PropTypes.string,
+		currentlyEditingMethod: PropTypes.shape( {
+			id: PropTypes.string,
+		} ),
+		method: PropTypes.shape( {
+			title: PropTypes.string.isRequired,
+			isSuggested: PropTypes.bool,
+			fees: PropTypes.string,
+			id: PropTypes.string,
+			informationUrl: PropTypes.string,
+		} ),
+		openPaymentMethodForEdit: PropTypes.func.isRequired,
+		savePaymentMethod: PropTypes.func.isRequired,
+	};
 
-	const onEditHandler = () => {
+	onEditHandler = () => {
+		const { method } = this.props;
+		const currentlyEditingId = this.props.currentlyEditingMethod &&
+			this.props.currentlyEditingMethod.id;
 		if ( currentlyEditingId === method.id ) {
-			onCancel( method );
+			this.onCancel( method );
 		} else {
-			onEdit( method );
+			this.onEdit( method );
 		}
 	};
 
-	return (
-		<ListItem>
-			<ListItemField>
-				{
-					method.isSuggested &&
-					(
-						<p className="payments__method-suggested">
-							{ translate( 'Suggested Method' ) }
+	onCancel = () => {
+		const { method } = this.props;
```

Comment:
Not sure its worth the bother to destructure method since you're only using it once below. Either that or destructure the other props you're using too maybe?